nipy/core/image/image_spaces.py

```diff
@@ -60,7 +60,7 @@
 It also works with nibabel images, which can only have xyz_affines:
 
 >>> import nibabel as nib
->>> nimg = nib.Nifti1Image(data, affine)
+>>> nimg = nib.Nifti1Image(data.astype('float'), affine)
```

Comment:
Sorry - remind me why we need to cast as float now?

Reply:
It's a quick hack. The problem is that `np.arange()` (used to generate synthetic data) produces `int64`s and now we error on int64 if we don't get an explicit `header` or `dtype` parameter.

* https://github.com/nipy/nibabel/issues/1089

Reply:
Worth a comment to that effect?

Reply:
I think a comment in a docstring would be more distracting than illuminating. Better solution, IMO, would be to update the `data = np.arange(...)` to be `data = np.arange(..., dtype='float32')`.

Reply:
Yes, that would also be fine.

Reply:
I pushed a follow-up commit that I think should be satisfactory.